docs/user_directory.md

```diff
@@ -1,49 +1,115 @@
-User Directory API Implementation
-=================================
+# User Directory API Implementation
 
-The user directory is currently maintained based on the 'visible' users
-on this particular server - i.e. ones which your account shares a room with, or
-who are present in a publicly viewable room present on the server.
+The user directory is maintained based on the 'visible' users of a homeserver -
+i.e. ones which are local to the server, ones which a local user shares a room
+with, or ones who are present in a publicly viewable room present on the server.
```

Comment:
Isn't that last one redundant? 

I think this is getting confused between which users the HS will index and which matches it will actually return to the user?

Reply:
I think you're right it is mixing concerns.

> The user directory is maintained based on the 'visible' users of a homeserver -
i.e. ones which are local to the server, and ones which a local user shares a room
with.

I think that's accurate? They don't need to be in public rooms.

Reply:
Yup, I think so.